### PR TITLE
Make current_tenant thread safe

### PR DESCRIPTION
**Pull request by sapanakothari**

### CODE REVIEW

**Comment by wireframe** [comment]
Thank you for your contribution!  The code addition is super straightforward, as expected.  The real trick is on the testing side.  My recommendation is to simplify the testcase such that the new thread attempts to overwrite the current tenant variable.  There are some good examples available here: https://stackoverflow.com/questions/6241384/how-do-i-manage-ruby-threads-so-they-finish-all-their-work

```ruby
describe '#current_tenant=` do
  it 'is threadsafe' do
    Multitenant.current_tenant = :old_tenant
    overwriter = Thread.new { Multitenant.current_tenant = :new_tenant }
    overwriter.join
    expect(Multitenant.current_tenant).to eq :old_tenant
  end
end
```

**Comment by sapanakothari** [comment]
> Thank you for your contribution! The code addition is super straightforward, as expected. The real trick is on the testing side. My recommendation is to simplify the testcase such that the new thread attempts to overwrite the current tenant variable. There are some good examples available here: https://stackoverflow.com/questions/6241384/how-do-i-manage-ruby-threads-so-they-finish-all-their-work
> 
> ```ruby
> describe '#current_tenant=` do
>   it 'is threadsafe' do
>     Multitenant.current_tenant = :old_tenant
>     overwriter = Thread.new { Multitenant.current_tenant = :new_tenant }
>     overwriter.join
>     expect(Multitenant.current_tenant).to eq :old_tenant
>   end
> end
> ```

Thankyou for your review and feedback. Changes are incorporated, kindly give another go.

File: spec/multitenant_spec.rb [hunk 1]
```diff
@@ -45,6 +45,15 @@ class Item < ActiveRecord::Base
     it { Multitenant.current_tenant == :foo }
   end
 
+  describe '#current_tenant=' do
+    it 'is threadsafe' do
+      Multitenant.current_tenant = :old_tenant
+      overwriter = Thread.new { Multitenant.current_tenant = :new_tenant }
+      overwriter.join
+      Multitenant.current_tenant == :old_tenant
```
Comment (wireframe) [inline spec/multitenant_spec.rb:53]:
```suggestion
      Multitenant.current_tenant.should == :old_tenant
```

Comment (sapanakothari) [reply spec/multitenant_spec.rb]: done

**Comment by wireframe** [comment]
Thank you @sapanakothari for your contribution!  This has been merged and released in the v1.0.0 release.

**Comment by sapanakothari** [comment]
Thanks @wireframe!

**Pull request merged**